fossology/fossology
# feat(newagent): new agent decision export import

## Description

Adding new agents (`decisionexporter` and `decisionimporter`) to export and import dump of FOSSology decisions as a JSON file.

The file will contain details about uploadtree, pfile, clearing decisions and events, copyright and ecc details, monkbulk scans, licenses and main licenses of upload.

The importer can use these information to recreate the license and copyright clearing on a new upload on a different server. This will help in merging various FOSSology instances as well.


### Changes

1. Create two new agents `decisionexporter` and `decisionimporter`.

## How to test

1. On an upload, perform license and copyright clearing. Also use monkbulk for license identification.
2. Create the JSON dump for the upload.
3. Upload the *exact same* upload in a different folder or even on a different server.
  - While performing the upload, select *only* nomos, ojo and monk agents.
4. Use the importer functionality to import the JSON dump.
5. Check if the new upload contains similar clearings as the original upload.

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2352"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
